src/content/blog/rust-based-javascript-linters-lack-typed-linting-right-now/index.mdx

```diff
@@ -0,0 +1,151 @@
+- Reimplement TypeScript's APIs in a native speed language
+- Speed Up TypeScript's APIs to native speed
+
+#### Option: Slowing Down to JavaScript Speed
+
+The performance hit option would likely slow the linters down to the point where they have no noticeable performance advantage compared to ESLint.
+👎
+
+That being said, if any native speed linter wants to do this, we in typescript-eslint would love to help.
+The [`@typescript-eslint/parser`](https://typescript-eslint.io/packages/parser) and [`@typescript-eslint/typescript-estree`](https://typescript-eslint.io/packages/typescript-estree) Node.js APIs are open source and as well documented as we've thought to write.
+We'd be happy to work with anybody who wants to use them, including spinning out standalone packages if that'd be useful.
+
+#### Option: Reimplementing TypeScript at Native Speed
+
+Reimplementing TypeScript at native speed is a tantalizing prospect for TypeScript users in general, not just linters.
+I know of three attempts: two actively under development ([Ezno](https://github.com/kaleidawave/ezno) and [`stc`](https://stc.dudy.dev)) and one older one ([TypeRunner](https://github.com/marcj/TypeRunner)).
+Both projects are very early stage and not likely to become production ready for a very long time.
+
+Keep in mind that re-implementing TypeScript in a new language is a herculean task.
+TypeScript's type checking has to deal with bizarrely complex edge cases around generic types, covariance, contravariance, and other terms most of us shudder to hear.
+
+> 💡 See [Ryan Cavanaugh's _Let's Make a Generic Inference Algorithm_ TypeScript Congress 2023 talk](https://portal.gitnation.org/contents/lets-make-a-generic-inference-algorithm) for an example of what TypeScript has to deal with.
+
+TypeScript is also a funded development team with contributions from its own programming language specialists and community contributors.
+Keeping up with its new versions is a Sisyphean task for any re-implementation.
+As impressive as Ezno and `stc` are, their long-term feasibility as standalone projects is precarious.
+
+> 💡 See [Matt Pocock's _Rewriting TypeScript in Rust? You'd have to be..._](https://www.totaltypescript.com/rewriting-typescript-in-rust) for more discussion with `stc`'s Donny.
+
+#### Option: Boosting TypeScript's APIs to Native Speed
+
+I think a more viable long-term option would be to find a way to get TypeScript's type checker to run at native speed.
+There are a couple possibilities:
+
+- Writing a tool that transpiles its source to a faster language such as Go or Rust
+- Pre-compiling and optimizing TypeScript like a binary
+
+Both of those options are difficult and will take some time to land.
+[Transpiling the checker to Go](https://kdy1.dev/2022-1-26-porting-tsc-to-go) was the original aim of what became `stc` before the [the project switched to the Rust re-implementation](https://kdy1.dev/2022-10-27-open-sourcing-stc#heading-ill-switch-back-to-the-rust-version).
```

Comment:
I don't know if there are any examples of a large project that has been successfully been transpiled between languages that could be mentioned here?

Go maybe possible, but I don't think it is possible to transpile to Rust because of Rust's reference system.

Reply:
Yeah I can't think of any 😞 but would have loved to.